File: apps/web/repositories/api/notification.repository.ts
// API Notification Repository implementation
import type { INotificationRepository } from "../types";
import type { NotificationOut } from "@/types/api";
import { apiClient } from "@/services/api-client";

export class ApiNotificationRepository implements INotificationRepository {
  public async list(params: {
    home_id: string;
    unread?: boolean;
    limit?: number;
    offset?: number;
  }): Promise<NotificationOut[]> {
    return apiClient.get("/notifications", {
      params: {
        home_id: params.home_id,
        unread: params.unread,
        limit: params.limit,
        offset: params.offset,
      },
    });
  }

  public async markRead(id: string): Promise<NotificationOut> {
    return apiClient.post(`/notifications/${id}/read`);
  }
}

File: apps/android/core/src/main/java/com/atech/core/data/NotificationRepository.kt
package com.atech.core.data

import com.atech.core.common.DomusResult
import com.atech.core.model.Notification
import com.atech.core.network.DomusHttp
import com.atech.core.network.safeApiCall
import io.ktor.client.request.get
import io.ktor.client.request.parameter
import io.ktor.client.request.post

/** Notifications. Mirrors `backend/notifications/router.py`. */
class NotificationRepository(private val http: DomusHttp) {
    private val base get() = "${http.config.apiBase}/notifications"

    suspend fun list(
        homeId: String? = null,
        unread: Boolean? = null,
        limit: Int = 50,
        offset: Int = 0,
    ): DomusResult<List<Notification>> =
        safeApiCall {
            http.client.get(base) {
                parameter("home_id", homeId)
                parameter("unread", unread)
                parameter("limit", limit)
                parameter("offset", offset)
            }
        }

    suspend fun markRead(notificationId: String): DomusResult<Notification> =
        safeApiCall { http.client.post("$base/$notificationId/read") }
}
